torch/export/_unflatten.py

```diff
@@ -0,0 +1,883 @@
+import abc
```

Comment:
Probably we can split the changes into two separate PRs: one for dynamo and another one for export.

Reply:
oops i think i accidentally committed this